publish-mqtt/src/main.rs

```diff
@@ -106,55 +110,110 @@ async fn main() -> Result<(), Box<dyn Error + Send + Sync>> {
     Ok(())
 }
 
+async fn connect_start_sensor<'a>(
+    bt_session: &'a BluetoothSession,
+    homie: &mut HomieDevice,
+    sensor_names: &HashMap<String, String>,
+    properties: Vec<Property>,
+    sensor: &BluetoothDevice<'a>,
+) -> Result<(), Box<dyn Error>> {
+    sensor.connect(CONNECT_TIMEOUT_MS)?;
+    start_notify_sensor(bt_session, &sensor)?;
+    let mac_address = sensor.get_address()?;
+    let node_id = mac_address.replace(":", "");
+    let node_name = sensor_names.get(&mac_address).unwrap_or(&mac_address);
+    homie
+        .add_node(Node::new(
+            node_id,
+            node_name.clone(),
+            "Mijia sensor".to_string(),
+            properties.to_vec(),
+        ))
+        .await?;
+    Ok(())
+}
+
 async fn requests(mut homie: HomieDevice) -> Result<(), Box<dyn Error>> {
     let sensor_names = hashmap_from_file(SENSOR_NAMES_FILENAME)?;
+    let max_connected_sensors = std::env::var("MAX_CONNECTED_SENSORS")
+        .ok()
+        .and_then(|val| val.parse::<usize>().ok())
+        .unwrap_or(DEFAULT_MAX_CONNECTED_SENSORS);
 
     homie.start().await?;
 
     let bt_session = &BluetoothSession::create_session(None)?;
     let device_list = scan(&bt_session).await?;
     let sensors = find_sensors(&bt_session, &device_list);
     print_sensors(&sensors, &sensor_names);
-    let (named_sensors, unnamed_sensors): (Vec<_>, Vec<_>) = sensors
+    let (named_sensors, mut unnamed_sensors): (Vec<_>, Vec<_>) = sensors
         .into_iter()
         .partition(|sensor| sensor_names.contains_key(&sensor.get_address().unwrap()));
-    println!("Connecting to named sensors first");
-    let mut connected_sensors = connect_sensors(&named_sensors);
-    println!("Connecting to unnamed sensors");
-    connected_sensors.extend(connect_sensors(&unnamed_sensors));
+    println!(
+        "{} named sensors, {} unnamed sensors",
+        named_sensors.len(),
+        unnamed_sensors.len()
+    );
 
-    let properties = vec![
+    let properties = [
         Property::new("temperature", "Temperature", Datatype::Float, Some("ºC")),
         Property::new("humidity", "Humidity", Datatype::Integer, Some("%")),
         Property::new("battery", "Battery level", Datatype::Integer, Some("%")),
     ];
-    for sensor in &connected_sensors {
-        let mac_address = sensor.get_address()?;
-        let node_id = mac_address.replace(":", "");
-        let node_name = sensor_names.get(&mac_address).unwrap_or(&mac_address);
-        homie.add_node(Node::new(
-            node_id,
-            node_name.clone(),
-            "Mijia sensor".to_string(),
-            properties.clone(),
-        ));
+
+    let mut sensors_to_connect: VecDeque<_> = named_sensors.into();
+    if sensors_to_connect.len() < max_connected_sensors {
```

Comment:
at some point, we will need to think about the case where named_sensors.len() > max_connected_sensors. I have made a card for it.